lib/active_merchant/billing/gateways/authorize_net_cim.rb

```diff
@@ -864,10 +864,12 @@ def commit(action, request)
         response_params['direct_response'] = parse_direct_response(response_params['direct_response']) if response_params['direct_response']
         transaction_id = response_params['direct_response']['transaction_id'] if response_params['direct_response']
 
-        Response.new(success, message, response_params,
-          :test => test_mode,
-          :authorization => transaction_id || response_params['customer_profile_id'] || (response_params['profile'] ? response_params['profile']['customer_profile_id'] : nil)
-        )
+        response_options = {}
+        response_options[:test] = test_mode
+        response_options[:authorization] = transaction_id || response_params['customer_profile_id'] || (response_params['profile'] ? response_params['profile']['customer_profile_id'] : nil)
+        response_options[:error_code] = response_params['messages']['message']['code'] unless success
```

Comment:
@timbeiko I made changes for this [request](http://developer.authorize.net/api/reference/index.html#payment-transactions-charge-a-customer-profile).

Look there the description for `messages` and `message` param: https://db.tt/qBsHeLWu
It says "one or more elements".

Further there is a Ruby code example there from official [library](https://github.com/AuthorizeNet/sample-code-ruby/blob/master/PaymentTransactions/charge-customer-profile.rb#L25-L32). As you see they rely on the same construction to get the error code.